### PR TITLE
WIP Extend replication protocol with ZenithStatusUpdate message

### PR DESCRIPTION
**Pull request by lubennikovaav**

### CODE REVIEW

File: src/backend/replication/walsender.c [hunk 1]
```diff
@@ -1902,6 +1907,90 @@ ProcessStandbyReplyMessage(void)
...
+	elog(LOG, "ProcessStandbyReplyMessage: writelsn %X/%X",
+					LSN_FORMAT_ARGS(writePtr));
+	elog(LOG, "ProcessStandbyReplyMessage: flushlsn %X/%X",
+					LSN_FORMAT_ARGS(flushPtr));
+	elog(LOG, "ProcessStandbyReplyMessage: applylsn %X/%X",
+					LSN_FORMAT_ARGS(applyPtr));
+}
+
+// This message is a zenith extension of postgres replication protocol
+// TODO unify with ParseZenithFeedbackMessage?
```
Comment (petuhovskiy) [inline src/backend/replication/walsender.c:1920]: +1 for reusing ParseZenithFeedbackMessage

File: src/backend/replication/walproposer.c [hunk 2]
```diff
@@ -190,6 +189,65 @@ WalProposerMain(Datum main_arg)
 	WalProposerStart();
 }
 
+static void
+ParseZenithFeedbackMessage(StringInfo reply_message, ZenithFeedback *zf)
```
[the remark this answers is not in the document]
Comment (lubennikovaav) [reply src/backend/replication/walproposer.c]: Done

File: src/backend/replication/walsender.c (hunk 1 above)
Comment (lubennikovaav) [reply src/backend/replication/walsender.c]: Done

File: src/backend/replication/walproposer.c [hunk 3]
```diff
@@ -2164,6 +2139,14 @@ AsyncReadMessage(Safekeeper *sk, AcceptorProposerMessage *anymsg)
 			msg->hs.ts = pq_getmsgint64_le(&s);
 			msg->hs.xmin.value = pq_getmsgint64_le(&s);
 			msg->hs.catalog_xmin.value = pq_getmsgint64_le(&s);
+			if (buf_size > APPENDRESPONSE_FIXEDPART_SIZE)
+			{
+				StringInfoData s;
```
Comment (petuhovskiy) [inline src/backend/replication/walproposer.c:2144]: We probably can reuse existing StringInfoData s, and replace `buf_size > APPENDRESPONSE_FIXEDPART_SIZE` with `s.len - s.cursor > 0`. Not sure how to keep AppendResponse backwards compatible in the future though.

File: src/backend/replication/walproposer.c [hunk 4]
```diff
@@ -1804,6 +1853,28 @@ GetAcknowledgedByQuorumWALPosition(void)
 	return responses[n_safekeepers - quorum];
 }
 
+static void
+GetLatestZentihFeedback(ZenithFeedback zf)
```
Comment (petuhovskiy) [inline src/backend/replication/walproposer.c:1857]: `ZenithFeedback *zf` ?

**Pull request merged**